### PR TITLE
fix: Allow Node Feature Discovery garbage collector to run on control-plane nodes

### PR DESCRIPTION
**Pull request by dlipovetsky**
**What problem does this PR solve?**:
With this change, all NFD components can run on a single node cluster.

**Which issue(s) this PR fixes**:
Fixes #https://jira.nutanix.com/browse/NCN-100706

**How Has This Been Tested?**:

**Special notes for your reviewer**:

### CODE REVIEW

File: charts/cluster-api-runtime-extensions-nutanix/templates/nfd/manifests/node-feature-discovery-configmap.yaml [hunk 1]
```diff
@@ -967,6 +967,11 @@ data:
           dnsPolicy: ClusterFirstWithHostNet
           securityContext: {}
           serviceAccountName: node-feature-discovery-gc
+          tolerations:
```
Comment (faiq) [inline charts/cluster-api-runtime-extensions-nutanix/templates/nfd/manifests/node-feature-discovery-configmap.yaml:970]: this is only going to happen when its deployed using cluster resource sets. we would also need to add this bit in the chart values with the helm addon provider.

Comment (faiq) [reply charts/cluster-api-runtime-extensions-nutanix/templates/nfd/manifests/node-feature-discovery-configmap.yaml:970]: copy these values to this file 

`charts/cluster-api-runtime-extensions-nutanix/templates/nfd/manifests/helm-addon-installation.yaml`

Comment (dlipovetsky) [reply charts/cluster-api-runtime-extensions-nutanix/templates/nfd/manifests/node-feature-discovery-configmap.yaml:970]: Got it. Thanks!

Comment (dlipovetsky) [reply charts/cluster-api-runtime-extensions-nutanix/templates/nfd/manifests/node-feature-discovery-configmap.yaml:970]: Done in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/722/commits/ad60625c904da207da7900862ac3c613bd7493a6

File: charts/cluster-api-runtime-extensions-nutanix/templates/nfd/manifests/helm-addon-installation.yaml [hunk 2]
```diff
@@ -27,4 +27,11 @@ data:
         - effect: NoSchedule
           key: node-role.kubernetes.io/control-plane
     ### <NFD-WORKER-CONF-END-DO-NOT-REMOVE>
+
+    gc:
+      tolerations:
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/master
```
Comment (supershal) [inline charts/cluster-api-runtime-extensions-nutanix/templates/nfd/manifests/helm-addon-installation.yaml:34]: `node-role.kubernetes.io/master` is deprecated. we should not expect it on the CAREN created cluster.

**Comment by jimmidyson** [comment]
From https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/723#issuecomment-2174406269:

> I wonder if NFD should also tolerate the CriticalAddonsOnly taint, as many other Addons do?

From what I can see, NFD is deployed with an empty `priorityClassName` so this wouldn't have any effect right now as the `CriticalAddonsOnly` taint is only applied temporarily when pre-empting lower priority pods.

So I guess the next question is should we set the `priorityClassName` on the NFD addon deployments and then tolerate `CriticalAddonsOnly`? 😅 I don't see any harm in it, but is it really critical?

File: charts/cluster-api-runtime-extensions-nutanix/templates/nfd/manifests/helm-addon-installation.yaml (hunk 2 above)
Comment (dlipovetsky) [reply charts/cluster-api-runtime-extensions-nutanix/templates/nfd/manifests/helm-addon-installation.yaml:34]: While that is true, we have 38 uses of that in CAREN. Let's replace it in a separate PR, please.

**Comment by dlipovetsky** [comment]
> But we should also see if we need to add `priorityClassName: system-cluster-critical` to the NFD pods?

I created a separate JIRA to review this for all Addons.

**Review by jimmidyson — APPROVED** [review]

**Review by mhrabovcin — APPROVED** [review]

**Pull request merged**